discovery/gossiper.go

```diff
@@ -518,6 +542,145 @@ func (d *deDupedAnnouncements) Emit() []lnwire.Message {
+				return err
+			}
+
+			// The first 33 bytes of the database key is
+			// the peer's public key.
+			peer, err := btcec.ParsePubKey(k[:33], btcec.S256())
+			if err != nil {
+				return err
+			}
+			t := msgTuple{peer, msg, k}
+
+			// Add the message to the slice, such that we
+			// can resend it after the database transaction
+			// is over.
+			msgsResend = append(msgsResend, t)
+			return nil
+		}); err != nil {
+			return err
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	// deleteMsg removes the message associated with the passed
+	// msgTuple from the messageStore.
+	deleteMsg := func(t msgTuple) error {
+		log.Debugf("Deleting message for chanID=%v from "+
+			"messageStore", t.msg.ChannelID)
+		if err := d.cfg.DB.Update(func(tx *bolt.Tx) error {
+			bucket := tx.Bucket(messageStoreKey)
+			if bucket == nil {
+				return fmt.Errorf("bucket " +
+					"unexpectedly did not exist")
+			}
+
+			return bucket.Delete(t.dbKey[:])
+		}); err != nil {
+			return fmt.Errorf("Failed deleting message "+
+				"from database: %v", err)
+		}
+		return nil
+	}
+
+	// We now iterate over these messages, resending those that we
+	// don't have the full proof for, deleting the rest.
+	for _, t := range msgsResend {
+		// Check if the full channel proof exists in our graph.
+		chanInfo, _, _, err := d.cfg.Router.GetChannelByID(
+			t.msg.ShortChannelID)
+		if err != nil {
+			// If the channel cannot be found, it is most likely
+			// a leftover message for a channel that was closed.
+			// In this case we delete it from the message store.
+			log.Warnf("unable to fetch channel info for "+
+				"chanID=%v from graph: %v. Will delete local"+
+				"proof from database",
+				t.msg.ChannelID, err)
+			if err := deleteMsg(t); err != nil {
+				return err
+			}
+			continue
+		}
+
+		// 1. If the full proof does not exist in the graph,
+		// it means that we haven't received the remote proof
+		// yet (or that we crashed before able to assemble the
+		// full proof). Since the remote node might think they
+		// have delivered their proof to us, we will resend
+		// _our_ proof to trigger a resend on their part:
+		// they will then be able to assemble and send us the
+		// full proof.
+		if chanInfo.AuthProof == nil {
+			err := d.sendAnnSigReliably(t.msg, t.peer)
+			if err != nil {
+				return err
+			}
+			continue
+		}
+
+		// 2. If the proof does exist in the graph, we have
+		// successfully received the remote proof and assembled
+		// the full proof. In this case we can safely delete the
+		// local proof from the database. In case the remote
+		// hasn't been able to assemble the full proof yet
+		// (maybe because of a crash), we will send them the full
+		// proof if we notice that they retry sending their half
+		// proof.
+		if chanInfo.AuthProof != nil {
+			log.Debugf("Deleting message for chanID=%v from "+
+				"messageStore", t.msg.ChannelID)
+			if err := d.cfg.DB.Update(func(tx *bolt.Tx) error {
```

Comment:
Seems like `deleteMsg` can be re-used here?

Reply:
Yep, that was the reason I created the method in the first place 🤦‍♂️